src/total-value-and-liquidity/total-value-and-liquidity.js

```diff
@@ -0,0 +1,130 @@
+const ethers = require('ethers');
+const BigNumber = require('bignumber.js');
+const { getJsonRpcUrl } = require('forta-agent');
+const RollingMathLib = require('rolling-math');
+
+// get createAlert and dataFields
+const { createAlert, dataFields } = require('./common');
+
+// load required shared types
+const contractAddresses = require('../../contract-addresses.json');
+
+const { LendingPool: LendingPoolAddr, ProtocolDataProvider: DataProviderAddr } = contractAddresses;
+const { abi: DataAbi } = require('../../interfaces/AaveProtocolDataProvider.json');
+const { abi: LendingPoolAbi } = require('../../interfaces/ILendingPool.json');
+
+// get config settings
+const {
+  totalValueAndLiquidity: Config,
+} = require('../../agent-config.json');
+
+// set up RPC provider
+const provider = new ethers.providers.WebSocketProvider(getJsonRpcUrl());
+
+// set up handle to Aave's LendingPool contract
+const LendingPool = new ethers.Contract(LendingPoolAddr, LendingPoolAbi, provider);
+const DataProvider = new ethers.Contract(DataProviderAddr, DataAbi, provider);
+
+// create rolling math object structure
+let rollingLiquidityData = {};
+
+// parses and returns data in a usable format
+async function parseData(dataPromise, reserve) {
+  // resolve our rpc call
+  const data = await dataPromise;
+
+  // calculate totals
+  const totalDebt = data.totalStableDebt.add(data.totalVariableDebt);
+  const totalValueLocked = totalDebt.add(data.availableLiquidity);
+
+  // create object with data to return
+  const parsedData = {};
+  dataFields.forEach((field) => {
+    parsedData[field] = data[field];
+  });
+
+  // add our custom fields
+  parsedData.reserve = reserve;
+  parsedData.totalDebt = totalDebt;
+  parsedData.totalValueLocked = totalValueLocked;
+
+  return parsedData;
+}
+
+function provideHandleBlock(RollingMath, config, lendingPool, dataProvider) {
+  rollingLiquidityData = {};
+
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    // override block number so we get data from the block in question
+    const override = { blockTag: blockEvent.blockNumber };
+
+    // get reserves and the current liquidity from this block
+    const reserves = await lendingPool.getReservesList({ ...override });
+
+    // create array containing a promise that returns the data for each reserve
+    const reserveData = [];
+    reserves.forEach((reserve) => {
+      // RPC call for per reserve data
+      const dataPromise = dataProvider.getReserveData(reserve, { ...override });
+      reserveData.push(
+        parseData(dataPromise, reserve),
+      );
+    });
+
+    // resolve our requests
+    const resolved = await Promise.all(reserveData);
```

Comment:
If any of the underlying promises rejects, this will fail fast.  Perhaps this should use the behavior we discussed this morning:
```suggestion
  const resolved = await Promise.all(reservedData.map((reservedDataPromise) => {
    return reservedDataPromise.catch((error) => console.log(error));
  }));
```
^ I did not run this to verify that it works, but the idea is the same.